lib/src/find.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// find.dart
+// Implementation of Find Functionality.
+//
+// 2023 July 11
+// Author: Rahul Gautham Putcha <rahul.gautham.putcha@intel.com>
+//
+
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+import 'package:rohd_hcl/src/count.dart';
+
+/// Find functionality
+///
+/// Takes in a [Logic] to find location of `1`s or `0`s
+class Find extends Module {
+  /// [_output] is output of Find (use index for accessing from outside Module)
+  late Logic _output;
+
+  /// [index] is an getter for output of Find
+  Logic get index => _output;
+
+  /// Find `1`s or `0`s
+  ///
+  /// Takes in filter search parameter [one], default [Find] `1`.
+  /// If [one] is `true` [Find] `1` else [Find] `0`.
+  ///
+  /// By default [Find] will look for first search parameter `1` or `0`.
+  /// If [n] is given, [Find] an [n]th search from first
+  /// occurance
+  Find(Logic bus, {bool one = true, Logic? n}) {
+    if (n != null) {
+      final oneHotList = <Logic>[];
+      for (var i = 0; i < bus.width; i++) {
+        if (one && i == 0) {
+          oneHotList.add(~bus[i] & n.eq(0));
+        } else {
+          final zeroCount = Count(bus.getRange(0, i + 1), countOne: one);
```

Comment:
let's name the variable something more flexible like just `count` or something, since this could be counting 1s or 0s